v2/README.md

```diff
@@ -0,0 +1,223 @@
+
+An early idea for CI is to run a simple script:
+
+1. clones the repo, run `go-licenses csv`.
+1. verifies if generated `licenses_info.csv` if up-to-date as the version in the repo.
+
+We might worry about flakiness, because various dependencies could be down
+temporarily. Another simpler idea is to let the script do:
+
+1. If `go.mod` has been updated, but not the license files.
+1. Fails and says you should update the license files.
+
+## Implementation Details
+
+Rough idea of steps in the two commands.
+
+`go-licenses csv` does the following to generate the `license_info.csv`:
+
+1. Load `go-licenses.yaml` config file, the config file can contain
+    * module name
+    * built binary local path
+    * module license overrides (path excludes or directly assign result license)
+1. All dependencies and transitive dependencies are listed by `go version -m <binary-path>`. When a binary is built with go modules, used module info are logged inside the binary. Then we parse go CLI result to get the full list.
+1. Scan licenses and report problems:
+    1. Use <github.com/google/licenseclassifier/v2> detect licenses from all files of dependencies.
+    1. Report an error if no license found for a dependency etc.
+1. Get license public URLs:
+    1. Get a dependency's github repo by fetching meta info like `curl 'https://k8s.io/client-go?go-get=1'`.
+    1. Get dependency's version info from go modules metadata.
+    1. Combine github repo, version and license file path to a public github URL to the license file.
+1. Generate CSV output with module name, license URL and license type.
+1. Report dependencies the tool failed to deal with during the process.
+
+`go-licenses save` does the following:
+
+1. Read from `license_info.csv` generated in `go-licenses csv`.
+1. Call [github.com/google/licenseclassifier](https://github.com/google/licenseclassifier) to get license type.
+1. Three types of reactions to license type:
+    * Download its notice and license for all types.
+    * Copy source folder for types that require redistribution of source code.
+    * Reject according to <https://github.com/google/licenseclassifier/blob/df6aa8a2788bdf5ac382148c2453a407a29819b8/license_type.go#L341>.
+
+## Credits
+
+go-licenses/v2 is greatly inspired by
+
+* [github.com/google/go-licenses](https://github.com/google/go-licenses) for the commands and compliance workflow
+* [github.com/mitchellh/golicense](https://github.com/mitchellh/golicense) for getting modules from binary
+* [github.com/uw-labs/lichen](https://github.com/uw-labs/lichen) for the vendored code to extract structured data from `go version -m` result.
+
+## Comparison with similar tools
+
+<!-- TODO(Bobgy): update this to a table -->
+
+* go-licenses/v2 was greatly inspired by [github.com/google/go-licenses](https://github.com/google/go-licenses), with the differences:
+  * go-licenses/v2 works better with go modules.
+    * no need to vendor dependencies.
```

Comment:
Thanks for the context.

I believe the concern does not apply because I am specifically talking about using go vendor is not necessary, but when running the save command full source folder is still copied like go-licenses v1.